components/infobox/wikis/starcraft2/infobox_person_player.lua

```diff
@@ -0,0 +1,539 @@
+			query = 'extradata, information, pagename',
+			limit = 10
+		})
+
+	if type(data[1]) == 'table' then
+		local rank1 = {}
+		local rank2 = {}
+		rank1['name'] = 'EPT ' .. (data[1].information or '') .. ' rank'
+		if type(data[1]) == 'table' then
+			local extradata = data[1].extradata
+			if extradata ~= nil then
+				rank1['rank'] = '[[' .. data[1].pagename .. '|#' .. extradata.rank .. ' (' .. extradata.points .. ' points)]]'
+			end
+		end
+		if type(data[2]) == 'table' then
+			local extradata = data[2].extradata
+			rank2['name'] = 'EPT ' .. (data[2].information or '') .. ' rank'
+			if extradata ~= nil then
+				rank2['rank'] = '[[' .. data[2].pagename .. '|#' .. extradata.rank .. ' (' .. extradata.points .. ' points)]]'
+			end
+		end
+		return rank1, rank2
+	end
+	return {}, {}
+end
+
+function StarCraft2Player._military(military)
+	if military and military ~= 'false' then
+		local display = military
+		military = string.lower(military)
+		local militaryCategory = ''
+		if String.Contains(military, 'starting') or String.Contains(military, 'pending') then
+			militaryCategory = '[[Category:Players waiting for Military Duty]]'
+			militaryStore = 'pending'
+		elseif String.Contains(military, 'ending') or String.Contains(military, 'started')
+			or String.Contains(military, 'ongoing') then
+				militaryCategory = '[[Category:Players on Military Duty]]'
+				militaryStore = 'ongoing'
+		elseif String.Contains(military, 'fulfilled') then
+			militaryCategory = '[[Category:Players expleted Military Duty]]'
+			militaryStore = 'fulfilled'
+		elseif String.Contains(military, 'exempted') then
+			militaryCategory = '[[Category:Players exempted from Military Duty]]'
+			militaryStore = 'exempted'
+		end
+
+		return display .. militaryCategory
+	end
+end
+
+function StarCraft2Player:getStatus(args)
+	if args.death_date then
+		statusStore = 'Deceased'
+	elseif args.retired then
+		statusStore = 'Retired'
+	elseif string.lower(args.role or 'player') ~= 'player' then
+		statusStore = 'not player'
+	end
+	return { store = statusStore }
+end
+
+function StarCraft2Player:addCustomCells(infobox, args)
+	local rank1, rank2
+	local yearsActive, activeCategory
+	if shouldStoreData and not statusStore then
+		rank1, rank2 = StarCraft2Player._getRank(self.pagename)
+		yearsActive, activeCategory = StarCraft2Player._getMatchupData(self.pagename)
+		infobox:categories(activeCategory)
+	end
+
+	local currentYearEarnings = earningsGlobal[tostring(currentYear)]
+	if currentYearEarnings then
+		currentYearEarnings = '$' .. mw.language.new('en'):formatNum(currentYearEarnings)
+	end
+
+	infobox:cell('Approx. Earnings ' .. currentYear, currentYearEarnings)
+	infobox:cell(rank1.name or 'Rank', rank1.rank)
+	infobox:cell(rank2.name or 'Rank', rank2.rank)
+	infobox:cell('Military Service', StarCraft2Player._military(args.military))
+	infobox:cell('Years active', yearsActive)
+
+	return infobox
+end
+
+function StarCraft2Player._getMatchupData(player)
```

Comment:
Same comment about this function as `_getEarningsMedalsData` above, this is really hard to read.